contrib/raftexample/kvstore.go

```diff
@@ -77,7 +77,7 @@ func (s *kvstore) readCommits(commitC <-chan *string, errorC <-chan error) {
 			if err := s.recoverFromSnapshot(snapshot.Data); err != nil {
 				log.Panic(err)
 			}
-			continue
+			return
```

Comment:
Is this correct. We might read snapshot from commitC in two cases:

1. restart with an existing snapshot
2. receive a snapshot from leader

return here will cause problem in case 2 from my understanding.